.github/workflows/tests.yml

```diff
@@ -57,21 +60,38 @@ jobs:
 
             - name: Run tests
               run: |
-                  yarn ${{ inputs.disable_coverage != 'true' && 'coverage' || 'test' }} \
+                  yarn test \
+                      --coverage=${{ env.ENABLE_COVERAGE && 'true' || 'false' }} \
                       --ci \
                       --reporters github-actions ${{ steps.metrics.outputs.extra-reporter }} \
                       --max-workers ${{ steps.cpu-cores.outputs.count }} \
                       --cacheDirectory /tmp/jest_cache
 
             - name: Upload Artifact
-              if: inputs.disable_coverage != 'true'
+              if: env.ENABLE_COVERAGE
               uses: actions/upload-artifact@v3
               with:
                   name: coverage
                   path: |
                       coverage
                       !coverage/lcov-report
 
+    skip_sonar:
+        name: Skip SonarCloud on merge_queue
+        if: github.event_name == 'merge_group' || inputs.disable_coverage == 'true'
```

Comment:
sadly can't reuse ENABLE_COVERAGE here as `env` isn't available to `job.<name>.if`